src/Makefile.am

```
@@ -707,6 +707,9 @@ uninstall-local: squid.conf.default
 	@$(SHELL) $(top_srcdir)/scripts/remove-cfg.sh "$(RM)" $(DESTDIR)$(DEFAULT_MIME_TABLE) $(srcdir)/mime.conf.default
 	@$(SHELL) $(top_srcdir)/scripts/remove-cfg.sh "$(RM)" $(DESTDIR)$(DEFAULT_CONFIG_FILE) squid.conf.default
 
+clean-local:
+	test -d "cf_gen.dSYM" && rm -rf "cf_gen.dSYM" || $(TRUE)
```

Automake [documentation](https://www.gnu.org/software/automake/manual/html_node/Clean.html) suggests a simpler variant that uses a recipe line prefix to [handle errors](https://www.gnu.org/software/make/manual/make.html#Errors-in-Recipes):

```suggestion
	-rm -rf cf_gen.dSYM
```

There is also no need to "quote" cf_gen.dSYM AFAICT (the above suggestion remove quotes).

I have not tested the above. If this simpler version works, we should use it IMO.

N.B. Squid already uses `-` recipe line prefixes in contrib/Makefile.am (at least).

---

The theoretical difference is that the existing line will not try to remove the directory if it's not there, where the proposed change will remove it unconditionally but then not fail if it can't (e.g. because it doesn't exist).

I'm not attached to the existing version and will accept the change to allow the PR to continue, but I doubt it's worth a blocking change request

---

> The theoretical difference is that the existing line will not try to remove the directory if it's not there, where the proposed change will remove it unconditionally but then not fail if it can't (e.g. because it doesn't exist).

Not exactly: AFAICT, `rm` does check whether the file is there (because it wants to get filesystem information about the file, very similar to what `test -d` would have to do), and `rm -rf` does not fail when the directory does not exist (it exits with zero status code instead).

There are, however, other, more important (IMO) differences. The proposed version is a lot simpler and does not introduce a yet another way of deleting things in Squid Automake files.
 
> I'm not attached to the existing version and will accept the change to allow the PR to continue, but I doubt it's worth a blocking change request

Thank you for accepting this change.